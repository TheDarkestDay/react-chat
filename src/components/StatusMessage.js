import React from 'react';
import PropTypes from 'prop-types';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';
import toMaterialStyle from 'material-color-hash';

import { User } from '../constants/shapes';
import generateLastActivityMessage from '../services/generate-last-activity-message';

const styles = () => ({
  messageWrapper: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: '8px',
    paddingBottom: '8px',
  },
});

const StatusMessage = (props) => {
  const {
    classes, content, createdAt, sender,
  } = props;

  return (
    <div className={classes.messageWrapper}>
      <Typography>
        <span style={{ color: toMaterialStyle(sender.username).backgroundColor }}>
          {sender.username}
        </span>
        <span>
          {content}
        </span>
      </Typography>
      <Typography variant="caption">
        {generateLastActivityMessage(createdAt)}
      </Typography>
    </div>
  );
};

StatusMessage.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  content: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  sender: User.isRequired,
};

export default withStyles(styles)(StatusMessage);
