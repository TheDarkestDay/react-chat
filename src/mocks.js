export const userMock = {
  _id: '1',
  username: 'test_user',
};

export const chatMock = {};

export const messageMock = {};
