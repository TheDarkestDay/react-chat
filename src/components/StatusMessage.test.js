/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StatusMessage from './StatusMessage';

import { userMock } from '../mocks';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<StatusMessage content="joined" createdAt="01-01-1990" sender={userMock} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should match snapshot', () => {
  const tree = renderer
    .create(<StatusMessage content="joined" createdAt="01-01-1990" sender={userMock} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
