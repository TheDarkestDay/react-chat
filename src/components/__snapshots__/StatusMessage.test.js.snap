// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="StatusMessage-messageWrapper-26"
>
  <p
    className="MuiTypography-root-27 MuiTypography-body1-36"
  >
    <span
      style={
        Object {
          "color": "#795548",
        }
      }
    >
      test_user
    </span>
    <span>
      joined
    </span>
  </p>
  <span
    className="MuiTypography-root-27 MuiTypography-caption-37"
  >
    2 days ago
  </span>
</div>
`;
